crates/sui-json/src/lib.rs

```diff
@@ -509,9 +509,22 @@ fn resolve_object_vec_arg(idx: usize, arg: &SuiJsonValue) -> Result<Vec<ObjectID
             }
             Ok(object_ids)
         }
+        JsonValue::String(s) if s.starts_with('[') && s.ends_with(']') => {
```

Comment:
Wondering: is the right fix here in SuiJSON, or in the CLI args parser?

I guess it would help me to understand all the places where we try to turn strings into SuiJSON.

Reply:
Good question! It seems to me like there is not custom CLI arg parsing as the CLI options are handled by a combination of clap and derived` Deserialize` (https://github.com/MystenLabs/sui/blob/main/crates/sui/src/client_commands.rs#L155)

I may be missing something, though, so perhaps @patrickkuo or @oxade could chime in here?

Reply:
I think we should put this logic in https://github.com/MystenLabs/sui/blob/278969ba8ba2c9e1dcc168441b127592ff70d49a/crates/sui-json/src/lib.rs#L301-L307

where we first convert the input string into SuiJsonValue

We don't distinguish CLI input with other string based inputs, I think @oxade have plan to improve this.

Reply:
I wonder what the advantage would be the advantage of moving it to the "constructor". The problem was only with parsing vector of objects - escaping vectors of (other) primitive types works just fine. For example:
```
sui client call --package $PKG --module entry_point_vector --function prim_vec_len --args \[42,7\] --gas-budget 500
----- Certificate ----
Transaction Hash: j/9iH8/GLqGy9b4Qz+Yl5BpOb7fzSkdN4uo/mv01RTU=
...
```

Reply:
@patrickkuo - any more feedback here?

Reply:
> I wonder what the advantage would be the advantage of moving it to the "constructor". The problem was only with parsing vector of objects - escaping vectors of (other) primitive types works just fine. For example:
> 
> ```
> sui client call --package $PKG --module entry_point_vector --function prim_vec_len --args \[42,7\] --gas-budget 500
> ----- Certificate ----
> Transaction Hash: j/9iH8/GLqGy9b4Qz+Yl5BpOb7fzSkdN4uo/mv01RTU=
> ...
> ```

Make sense